### PR TITLE
State transition diagram.

### PR DESCRIPTION
**Pull request by istoica**
## What do these changes do?

Trivial PR containing some method and variable name changes for increased clarity, plus a description of the state transition diagram for tasks.

### CODE REVIEW

**Comment by robertnishihara** [comment]
Nice! It looks like a bunch of other changes were unintentionally included in this PR. The only file that we want to change here is `task_states.rst, right?

**Comment by istoica** [comment]
@robertnishihara There are a few name changes and comment updates in the other files. All of files are intended to be part of the PR.

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -142,12 +142,6 @@ ray::Status NodeManager::RegisterGcs() {
     ClientAdded(data);
   };
   gcs_client_->client_table().RegisterClientAddedCallback(node_manager_client_added);
-  // Register a callback on the client table for removed clients.
-  auto node_manager_client_removed = [this](
-      gcs::AsyncGcsClient *client, const UniqueID &id, const ClientTableDataT &data) {
-    ClientRemoved(data);
-  };
-  gcs_client_->client_table().RegisterClientRemovedCallback(node_manager_client_removed);
```
Comment (robertnishihara) [inline src/ray/raylet/node_manager.cc]: this change is accidental I think

File: src/ray/raylet/node_manager.cc [hunk 2]
```diff
@@ -209,15 +203,6 @@ void NodeManager::Heartbeat() {
 
 void NodeManager::ClientAdded(const ClientTableDataT &client_data) {
   ClientID client_id = ClientID::from_binary(client_data.client_id);
-
-  // Make sure the client hasn't already been removed.
-  if (removed_clients_.find(client_id) != removed_clients_.end()) {
-    // This client has already been removed, so don't do anything.
-    RAY_LOG(INFO) << "The client " << client_id << " has already been removed, so it "
-                  << "can't be added. This is very unusual.";
-    return;
```
Comment (robertnishihara) [inline src/ray/raylet/node_manager.cc]: this change is accidental I think

File: src/ray/raylet/node_manager.cc [hunk 3]
```diff
@@ -255,36 +240,6 @@ void NodeManager::ClientAdded(const ClientTableDataT &client_data) {
   remote_server_connections_.emplace(client_id, std::move(server_conn));
 }
 
-void NodeManager::ClientRemoved(const ClientTableDataT &client_data) {
-  const ClientID client_id = ClientID::from_binary(client_data.client_id);
```
Comment (robertnishihara) [inline src/ray/raylet/node_manager.cc]: this change is accidental I think

File: src/ray/raylet/node_manager.h [hunk 4]
```diff
@@ -62,8 +62,6 @@ class NodeManager {
   /// Methods for handling clients.
   /// Handler for the addition of a new GCS client.
   void ClientAdded(const ClientTableDataT &data);
-  /// Handler for the removal of a GCS client.
-  void ClientRemoved(const ClientTableDataT &client_data);
```
Comment (robertnishihara) [inline src/ray/raylet/node_manager.h]: this change is accidental I think

File: src/ray/raylet/node_manager.h [hunk 5]
```diff
@@ -135,9 +133,6 @@ class NodeManager {
   /// The lineage cache for the GCS object and task tables.
   LineageCache lineage_cache_;
   std::vector<ClientID> remote_clients_;
-  /// A set of all of the remote clients that have been removed. In principle,
-  /// this could grow unbounded.
-  std::unordered_set<ClientID> removed_clients_;
```
Comment (robertnishihara) [inline src/ray/raylet/node_manager.h]: this change is accidental I think

File: src/ray/raylet/node_manager_test.cc [hunk 6]
```diff
@@ -0,0 +1,79 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
```
Comment (robertnishihara) [inline src/ray/raylet/node_manager_test.cc]: let's remove this file

File: src/ray/raylet/scheduler_test.cc [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
```
Comment (robertnishihara) [inline src/ray/raylet/scheduler_test.cc]: let's remove this file

File: src/ray/raylet/design_docs/task_states.rst [hunk 8]
```diff
@@ -0,0 +1,36 @@
+A task can be in one of the following states:
```
Comment (robertnishihara) [inline src/ray/raylet/design_docs/task_states.rst]: Let's move this to the top level directory under `design_docs/task_states.rst`.

Comment (pcmoritz) [reply src/ray/raylet/design_docs/task_states.rst]: how about we put it into a subdirectory of ray/doc? Like ray/doc/design for example

Comment (robertnishihara) [reply src/ray/raylet/design_docs/task_states.rst]: On the other hand, if we want them to be in the documentation page, then we need them to be under `doc/source/`, but that can be done later.

Comment (robertnishihara) [inline src/ray/raylet/design_docs/task_states.rst]: Can you add something like this to the top of the file? (this is normal for rst files)

```
Task States
===========
```

File: src/ray/raylet/design_docs/task_states.rst [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+- Placeable: the task is ready to be placed at the node where is going to be
```
Comment (robertnishihara) [inline src/ray/raylet/design_docs/task_states.rst]: May want to make `Placeable:` bold (and same for the other states). You can do this with `**Placeable:**`.

**Pull request closed without merging**